src/Board.hs

```diff
@@ -60,14 +62,14 @@ scrambleBoard board@(Board (Dimensions width height) _) startingBoard gen =
                in verifyBoard newBoard gen iterations
           verifyBoard board gen iterations
               | board == startingBoard = scrambleBoardLoop board gen iterations
-              | otherwise              = scrambleBoardLoop board gen (pred iterations)
+              | otherwise              = scrambleBoardLoop board gen (iterations - 1)
 
 -- | Randomly generate a new board, given dimensions.
 newBoard :: Dimensions -> IO Board
 newBoard d = do
     newGen <- R.newStdGen
-    return $ scrambleBoard startingBoard startingBoard newGen
-    where startingBoard = calculateStartingBoard $ Board d []
+    return (scrambleBoard startingBoard startingBoard newGen)
+    where startingBoard = calculateStartingBoard (Board d [])
```

Comment:
I have a habit of, when I'm editing code, changing it to conform with my style. You can disregard style changes like this; I don't mean to suggest that my style preferences are better than yours!